service/telemetry.go

```diff
@@ -225,15 +225,21 @@ func (tel *telemetryInitializer) initOpenCensus(cfg telemetry.Config, telAttrs m
 	return pe, nil
 }
 
-func (tel *telemetryInitializer) initOpenTelemetry() (http.Handler, error) {
+func (tel *telemetryInitializer) initOpenTelemetry(attrs map[string]string) (http.Handler, error) {
 	// Initialize the ocRegistry, still used by the process metrics.
 	tel.ocRegistry = ocmetric.NewRegistry()
 
 	exporter := otelprom.New()
 	tel.mp = sdkmetric.NewMeterProvider(sdkmetric.WithReader(exporter))
 
 	registry := prometheus.NewRegistry()
-	if err := registry.Register(exporter.Collector); err != nil {
+	promLabels := make(prometheus.Labels)
+	for k, v := range attrs {
+		promLabels[sanitizePrometheusKey(k)] = v
+	}
```

Comment:
I think the "attrs" are actually "Resource" attributes and they should be configured accordingly to the `MeterProvider`?

Reply:
Agreed, but right now resource attributes are not exported by the prometheus exporter. To keep the otel prometheus metrics consistent with the existing oc prometheus we need that.

Reply:
I've added `attrs` as `Resource` attributes to the sdk (which will be useful if we want to support other formats) but if we want to keep the parity with the existing oc metrics we still need to add them as prometheus constant labels.

If you think we should drop it, i.e., losing `service_instance_id` and `service_version` from our prom metrics, I can update the pr.

Reply:
See my next comment, for the moment add them, but let's track this bug. I expect this to be fixed for prometheus soon correct.